dbms/src/Flash/Mpp/MPPTunnelSet.cpp

```diff
@@ -30,79 +30,48 @@ void checkPacketSize(size_t size)
         throw Exception(fmt::format("Packet is too large to send, size : {}", size));
 }
 
+TrackedMppDataPacketPtr serializePacket(tipb::SelectResponse & response)
+{
+    auto tracked_packet = std::make_shared<TrackedMppDataPacket>();
+    tracked_packet->serializeByResponse(response);
+    checkPacketSize(tracked_packet->getPacket().ByteSizeLong());
+    return tracked_packet;
+}
 } // namespace
 
 template <typename Tunnel>
-void MPPTunnelSetBase<Tunnel>::clearExecutionSummaries(tipb::SelectResponse & response)
+void MPPTunnelSetBase<Tunnel>::sendExecutionSummary(tipb::SelectResponse & response)
 {
-    /// can not use response.clear_execution_summaries() because
-    /// TiDB assume all the executor should return execution summary
-    for (int i = 0; i < response.execution_summaries_size(); i++)
-    {
-        auto * mutable_execution_summary = response.mutable_execution_summaries(i);
-        mutable_execution_summary->set_num_produced_rows(0);
-        mutable_execution_summary->set_num_iterations(0);
-        mutable_execution_summary->set_concurrency(0);
-    }
+    RUNTIME_CHECK(!tunnels.empty());
+    // for execution summary, only need to send to one tunnel.
+    tunnels.back()->write(serializePacket(response));
```

Comment:
Why change the original logic which always send summary to the first tunnel?

Reply:
No special reason, I'll revert back to the original logic..